rotkehlchen/balances/historical.py

```diff
@@ -120,6 +121,67 @@ def get_assets_amounts(
 
         return amounts, last_event_id
 
+    def get_historical_netvalue(
+            self,
+            from_ts: Timestamp,
+            to_ts: Timestamp,
+    ) -> tuple[dict[Timestamp, FVal], list[tuple[str, Timestamp]]]:
+        """Calculates historical net worth per day within the given time range.
+
+        Uses asset balances per day combined with historical prices to calculate
+        the total net worth in the user's main currency for each day.
+
+        May raise:
+            - NotFoundError if no events exist in the specified time period.
+
+        Returns:
+            - A mapping of timestamps to net worth in main currency for each day
+            - A list of (asset id, timestamp) tuples where price data was missing
+        """
+        events, main_currency = self._get_events_and_currency(from_ts=from_ts, to_ts=to_ts)
+        if len(events) == 0:
+            raise NotFoundError(f'No historical data found within {from_ts=} and {to_ts=}.')
+
+        current_balances: dict[Asset, FVal] = defaultdict(FVal)
+        daily_balances: dict[Timestamp, dict[Asset, FVal]] = defaultdict(lambda: defaultdict(FVal))
+        for event in events:
+            if (
+                (direction := event.maybe_get_direction()) is None or
+                direction == EventDirection.NEUTRAL
+            ):
+                continue
+
+            if direction == EventDirection.IN:
+                current_balances[event.asset] += event.balance.amount
+            elif direction == EventDirection.OUT:
+                current_balances[event.asset] -= event.balance.amount
+
+            daily_balances[timestamp_to_daystart_timestamp(ts_ms_to_sec(event.timestamp))] = current_balances.copy()  # noqa: E501
+
+        # Calculate net worth for each day using cached prices
+        net_worth_per_day: dict[Timestamp, FVal] = {}
+        missing_price_points: list[tuple[str, Timestamp]] = []
+        for day_ts, asset_balances in daily_balances.items():
+            day_total = ZERO
+            for asset, balance in asset_balances.items():
+                if balance == ZERO:
+                    continue
+
+                if (result := GlobalDBHandler.get_historical_price(
+                    from_asset=asset,
+                    timestamp=day_ts,
+                    to_asset=main_currency,
+                    max_seconds_distance=DAY_IN_SECONDS,
+                )) is None:  # Skip this asset's value for the day's total if we can't price it # noqa: E501
+                    missing_price_points.append((asset.identifier, day_ts))
+                    continue
```

Comment:
each one of those is a trip to the globaldb... I fear this can be really slow. Can't we batch it somehow for a whole day? So create a function that queries several assets at once. Imagine you have 100 tokens and you do this for 720 days. That is 720 * 120 queries to the db when you can do only 720.